### ISSUE TITLE
Error % shown in metrics detail page and in Services List page is different

### ISSUE THREAD

**Issue by pranay01** [SigNoz/signoz#1259]
Sometimes the values shown in `Error Percentage` panel in application detail page doesn't match with the Error % shown in services list page for that application

**Comment by srikanthccv** [comment]
Query service /overview error calculation - based on the span table entries which have `hasError` set to true i.e `http.status_code` >= 400 or `rpc.grpc.status_code` >= 2 or span status == 2 ( == `ERROR`).

Promql expression for error calculation - >= 400 && < 500 are not considerd which can introduce the variation. 
```sql
max(sum(rate(signoz_calls_total{service_name="route", span_kind="SPAN_KIND_SERVER", status_code="STATUS_CODE_ERROR"}[1m]) OR rate(signoz_calls_total{service_name="route", span_kind="SPAN_KIND_SERVER", http_status_code=~"5.."}[1m]))*100/sum(rate(signoz_calls_total{service_name="route", span_kind="SPAN_KIND_SERVER"}[1m]))) < 1000 OR vector(0)
```

**Comment by ankitnayan** [comment]
The application list page uses below query at `GetServices` method
```
SELECT serviceName, count(*) as numErrors FROM %s.%s WHERE timestamp>='%s' AND timestamp<='%s' AND kind='2' AND (statusCode>=500 OR statusCode=2)
```

The application overview page uses promql that you mentioned to get error data
```
 max(sum(rate(signoz_calls_total{service_name="route", span_kind="SPAN_KIND_SERVER", status_code="STATUS_CODE_ERROR"}[1m]) OR rate(signoz_calls_total{service_name="route", span_kind="SPAN_KIND_SERVER", http_status_code=~"5.."}[1m]))*100/sum(rate(signoz_calls_total{service_name="route", span_kind="SPAN_KIND_SERVER"}[1m]))) < 1000 OR vector(0)
```


The below logic you mentioned is used in the same `overview` page and is not used for error calculation in the overview page.
> hasError set to true i.e http.status_code >= 400 or rpc.grpc.status_code >= 2 or span status == 2 ( == ERROR)

@palashgdev can you confirm this?

**Comment by palashgdev** [comment]
@ankitnayan i am not clear with the comment

**Comment by ankitnayan** [comment]
@srikanthccv I confirmed from @palashgdev that `overview` api is not used to create error percentage chart.

**Comment by ankitnayan** [comment]
@pranay01 I see this happen when the metrics is not being stored due to `otel-collector-metrics` crashing. In such case, `error %` and `rps` both charts plots are empty or 0. I don't think this is a case of data sanity. As the application list page fetches data from the trace data whereas the `rps` and `error %` in application overview page fetches data from the metrics api, this scenario will be repeated if traces are being ingested properly but metric ingestion is failing.

 If someone else notices this, we can re-open the issue

**Comment by srikanthccv** [comment]
>@srikanthccv I confirmed from @palashgdev that overview api is not used to create error percentage chart.

Ok, I was confused then. A week back we fixed an issue with error value here https://github.com/SigNoz/signoz/pull/1229. That could have been one of the possible reasons for the disparity.

**Comment by ankitnayan** [comment]
Yes, that could have caused that. Thanks for fixing that !

**Comment by ankitnayan** [comment]
Another case was observed where both `error` and `rps` were not having data.

**Comment by srikanthccv** [comment]
@ankitnayan any chance you were looking at the external calls tab when you made the above comment? because that's a legitimate case we know not working for rpc systems but no the service incoming rps. We could see the data exist for each service yesterday.

**Comment by ankitnayan** [comment]
@srikanthccv I noticed no data in error and rps but the apis from traces were showing results. It's not about data inconsistency but it might have more to do about data not available as metrics. We can close this issue if you think it's okay.

**Comment by srikanthccv** [comment]
Since that is different issue, I am closing this. What we have known from one of users is that they didn't have data ingested for `signoz_*` metrics, which happens when the collectors from node not sending data. If we come across new users who have the same issue but for different reasons we can create another issue.

**Comment by srikanthccv** [comment]
re-opening since we suspect some metrics could be dropped because of issues with cache eviction/addition mechanism or size.

**Comment by srikanthccv** [comment]
@ankitnayan I believe we are sorted from the calc side issue. You should be able to paly around with grpc/http-client/server data with the latest staging app and see the parity b/w list page and detailed page.

**Comment by srikanthccv** [comment]
If the issue still persists after this then it's the writer dropping the metrics. I suspect the issue there is more likely an synchronization related than the size of cache. However, we should try to reproduce it and fix.

**Comment by srikanthccv** [comment]
I am going to close this for now since we release a fix in 0.10.0. If any data dropping issues noted will be addressed in separate issue.
